### PR TITLE
Pm2 config

### PR DESCRIPTION
**Pull request by riteshsp2000**
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this
repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**
      (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch**
      (left side). Also you should start _your branch_ off _default branch_.
- [x] Check the commit's or even all commits' message styles matches our
      requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit
      test.
- [x] I have added necessary documentation (if appropriate)

### Description
Configure pm2 for development and production.

Please describe your pull request.

❤️Thank you!

### Post merge checklist

- [ ] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to
      this repository.
- [ ] If you are a new contributor, ping in the thread and one of the
      maintainers will add you to all-contributors list.

### CODE REVIEW

**Review by abhibhaw — CHANGES_REQUESTED** [review]
Don't we need to change the build commands in the workflows after this pm2 configuration?

File: ecosystem.config.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+module.exports = {
+  apps: [
```
Comment (abhibhaw) [inline ecosystem.config.js:2]: Also we need 3 apps configurations, for staging, prod and PR

Comment (riteshsp2000) [reply ecosystem.config.js:2]: yes yes, I am working on this finding some suitable settings for the corresponding.

**Comment by riteshsp2000** [comment]
> Don't we need to change the build commands in the workflows after this pm2 configuration?

Yes, we do need to update them, How about after this is created, you create a PR setting up the workflows for the remaining environments. I'll keep the original run/build commands as well so that this wont affect the current staging after merger.

**Comment by abhibhaw** [comment]
Yo sounds great 🦾

**Comment by abhibhaw** [comment]
You may need to change the app name as well, I am quite unsure about the spacing in the app name 🤔

**Comment by riteshsp2000** [comment]
> You may need to change the app name as well, I am quite unsure about the spacing in the app name 🤔

nah nah.... I'm pretty sure we can name with a space.

File: ecosystem.config.js (hunk 1 above)
Comment (DesignrKnight) [reply ecosystem.config.js:2]: Why do you need three configs for PM2 ecosystem config? You can have same app config used in all three environments na?

File: package.json [hunk 2]
```diff
@@ -4,9 +4,21 @@
...
+    "prettier": "npx prettier --write .",
+    "prettier:check": "npx prettier --check .",
+    "lint:fix": "npx eslint --fix '**/*.ts'",
+    "lint:find": "npx eslint '**/*.ts",
+    "transpile:dev": "tsc --watch",
+    "transpile:prod": "tsc",
+    "start:dev": "npx pm2 startOrGracefulReload ecosystem.config.js --env development",
+    "start:stage": "pm2 startOrGracefulReload ecosystem.config.js --env staging",
+    "start:prod": "pm2-runtime ecosystem.config.js --env production",
+    "stop": "npx pm2 stop ecosystem.config.js && npx pm2 delete ecosystem.config.js",
```
Comment (DesignrKnight) [inline package.json:16]: calling a command stop and then deleting ain't good

File: ecosystem.config.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      error_file: './logs/pm2-err.log',
+      out_file: './logs/pm2-out.log',
+      log_file: './logs/pm2-app.log',
+      combine_logs: false,
+      kill_timeout: 2000,
+      min_uptime: '1m',
+      max_restarts: 5,
+      restart_delay: 1000,
+      autorrestart: true,
+    },
```
Comment (DesignrKnight) [inline ecosystem.config.js:20]: There is a thing known as an exponential back-off restart. Use that.

I see that is watch is false here, what do you plan for development phase?

File: ecosystem.config.js (hunk 1 above)
Comment (abhibhaw) [reply ecosystem.config.js:2]: We have different env and app name for all 3. And that env's are subject to change during development time so was planning to have everything separate for now. Can keep one after the first production release.

File: ecosystem.config.js (hunk 3 above)
Comment (abhibhaw) [reply ecosystem.config.js:20]: Planning not to use pm2 during dev but will include this if you suggest.

**Pull request closed without merging**